NPE when publishing to an incorrect URL

I was testing the publication of a plugin created with the command `nextflow plugin create` . It configures the registry to `https://nf-plugins-registry.dev-tower.net/api`, which doesn't exist. The command `./gradlew releasePlugin` returns the following NPE instead of an error.

```
Caused by: java.lang.NullPointerException: Cannot invoke "io.nextflow.gradle.registry.RegistryClient$ErrorResponse.getType()" because "err" is null
        at io.nextflow.gradle.registry.RegistryClient.publish(RegistryClient.groovy:37)
        at io.nextflow.gradle.registry.RegistryUploadTask.run(RegistryUploadTask.groovy:34)
```

--- a/src/main/groovy/io/nextflow/gradle/registry/RegistryClient.groovy
+++ b/src/main/groovy/io/nextflow/gradle/registry/RegistryClient.groovy
@@ -1,7 +1,6 @@
 package io.nextflow.gradle.registry
 
 import com.google.gson.Gson
-import com.google.gson.JsonParseException
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
 import org.apache.http.client.methods.CloseableHttpResponse
@@ -38,7 +37,7 @@ class RegistryClient {
              def rep = http.execute(req)) {
 
             if (rep.statusLine.statusCode != 200) {
-                throw new RuntimeException(getErrorMessage(rep))
+                throw new RegistryPublishException(getErrorMessage(rep))
             }
         } catch (ConnectException e) {
             throw new RuntimeException("Unable to connect to plugin repository: (${e.message})")
@@ -50,23 +49,9 @@ class RegistryClient {
         if( rep.entity ) {
             final String entityStr = EntityUtils.toString(rep.entity)
             if (entityStr) {
-                try {
-                    def err = gson.fromJson(entityStr, ErrorResponse)
-                    if( err )
-                        return "$message - Error type: ${err.type}, message: ${err.message}".toString()
-                } catch( JsonParseException e ) {
-                    log.debug("Exception parsing error response: $e.message")
-                }
                 return "$message - $entityStr".toString()
             }
         }
         return message.toString()
-    }
-
-    // ----------------------------------------------------------------------------
-
-    private static class ErrorResponse {
-        String type
-        String message
     }
 }

--- a/src/main/groovy/io/nextflow/gradle/registry/RegistryPublishException.groovy
+++ b/src/main/groovy/io/nextflow/gradle/registry/RegistryPublishException.groovy
@@ -1,0 +1,10 @@
+package io.nextflow.gradle.registry
+
+/**
+ * Custom exception class for registry publish task
+ */
+class RegistryPublishException extends Exception{
+    RegistryPublishException(String s) {
+        super(s)
+    }
+}

